pkg/controller/scheduledjob/controller.go

```diff
@@ -203,8 +203,7 @@ func SyncOne(sj batch.ScheduledJob, js []batch.Job, now time.Time, jc jobControl
 		return
 	}
 	if sj.Spec.ConcurrencyPolicy == batch.ReplaceConcurrent {
-		for i := range sj.Status.Active {
-			j := sj.Status.Active[i]
```

Comment:
Since you are doing a single update with all the deletions, why not use a separate slice of object references that will hold the new references and update the sj with it before the update? It's a bigger change than this but also feels saner:)